.editorconfig

```diff
@@ -0,0 +1,35 @@
+# EditorConfig is awesome: http://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+# Unix-style newlines with a newline ending every file
+[*]
+end_of_line = lf
+insert_final_newline = true
+charset = utf-8
+indent_style = space
+indent_size = 2
```

Comment:
I don't think the indent size can be set generally for all files, notably test files, as many will probably be using four spaces (or whatever their syntax prefers). You unset them later for syntax test files, but why set it indentation settings globally at all then? What is the purpose of this specification?

Reply:
The global settings are perhaps force of habit. Normally I'd say the purpose is consistency across a code base, but in this particular case I suppose the primary purpose is actually to ensure that mutative settings are disabled for the files that do have strict requirements. This repo is also not a typical collection of files and contains many very different languages.

So the global/default settings can be removed, or perhaps just reduced to `charset`.